content/docs/gke/customizing-gke.md

```diff
@@ -7,56 +7,104 @@ weight = 2
+**skip the `kfctl apply` command** and run the **`kfctl build`** command 
+instead, as  described in that step. Now you can edit the configuration files 
+before deploying Kubeflow.
+
+## Customizing an existing deployment
+
+You can also customize an existing Kubeflow deployment. In that case, this 
+guide assumes that you have already followed the guide to 
+[deploying Kubeflow on GCP](/docs/gke/deploy/deploy-cli/) and have deployed
+Kubeflow to a GKE cluster.
+
 ## Before you start
 
-This guide assumes you have already set up Kubeflow with GKE. If you haven't done
-so, follow the guide to [deploying Kubeflow on GCP](/docs/gke/deploy/).
+This guide assumes the following settings: 
 
-## Customizing Kubeflow
+* The `${KFAPP}` environment variable contains the name
+  of your Kubeflow application directory, as described in the guide to
+  [deploying Kubeflow on GCP](/docs/gke/deploy/deploy-cli/):
 
-You can use [kustomize](https://kustomize.io/) to customize Kubeflow.
+  ```
+  export KFAPP=<your choice of application directory name>
+  ```
+
+* The `${CONFIG}` environment variable contains the path to a kfctl 
+  configuration file, as described in the section on [preparing your environment 
+  for deployment](/docs/gke/deploy/deploy-cli/#prepare-environment):
 
-The deployment process is divided into two steps, **generate** and **apply**, so that you can
-modify your deployment before actually deploying.
+  ```
+  export CONFIG="{{% config-uri-gcp-iap %}}"
+  ```
+
+    Or:
 
-To customize GCP resources (such as your Kubernetes Engine cluster), you can modify the deployment manager configs in **${KFAPP}/gcp_config**.
+  ```
+  export CONFIG="{{% config-uri-gcp-basic-auth %}}"
+  ```
+
+* All commands below start from the parent directory that contains your 
+  `${KFAPP}` directory:
+
+  ```
+  cd <the parent directory of your KFAPP directory>
+  ```
 
-Many changes can be applied to an existing configuration in which case you can run:
+## Customizing GCP resources
+
+To customize GCP resources, such as your Kubernetes Engine cluster, you can 
+modify the Deployment Manager configuration settings in `${KFAPP}/gcp_config`.
+
+After modifying your existing configuration, run the following command to apply
+the changes:
 
 ```
 cd ${KFAPP}
-kfctl apply platform
+kfctl apply -V -f ${CONFIG}
 ```
 
-or using Deployment Manager directly:
+Alternatively, you can use Deployment Manager directly:
 
 ```
 cd ${KFAPP}/gcp_config
 gcloud deployment-manager --project=${PROJECT} deployments update ${DEPLOYMENT_NAME} --config=cluster-kubeflow.yaml
 ```
 
-  * **PROJECT** Name of your GCP project. You could find it in `${KFAPP}/app.yaml`.
-  * **DEPLOYMENT_NAME** Name of your Kubeflow app. You could also find it in `${KFAPP}/app.yaml`. 
-    In specific, `.metadata.name`
+  * **PROJECT** The ID of your GCP project. You can find the project ID in 
+    your `${KFAPP}/app.yaml` file, as the value for the `project` key.
```

Comment:
${KFAPP}/app.yaml should change hopefully by 0.7.0

It should be

${KFAPPDIR}/${file_name}

Where filename is just the filename of the file they applied  e.g.

if they do

kfctl apply -f https://github.com/kfctl_gcp_iap.yaml

Then the command here would be

kfctl apply -f kfctl_gcp_iap.yaml

Or they could specify the full path.

kfctl apply -f ${KFAPPDIR}/kfctl_gcp_iap.yaml

Another way to think of it and possibly explain it is that users should always do

kfctl apply -f ${KFAPPDIR}/${KF_CONFIG_FILE}

The "one exception" is if the file is a remote URI e.g. 
kfctl apply -f https://....

In this case when a user runs kfctl apply -f we will copy the remote file to ${KFAPPDIR}/

All subsequent commands should then be kfctl apply -f ${KFAPPDIR}/${KFFILE}

Reply:
Thanks @jlewi - it seems this change is not certain for v0.7? If it's not certain, then let's put this change into a separate issue, which we can address when the change happens. LMK and I'm happy to create the issue.

Reply:
I think it needs to land before we finalize the RC. Its a P0 bug.
https://github.com/kubeflow/kfctl/issues/54

Reply:
I feel more comfortable with updating the docs to reflect the desired syntax. From a 0.7 perspective if the file gets renamed to "app.yaml" that's a P0 bug. So I'd rather have the docs be consistent with that expectation as opposed to risk setting an expectation that "app.yaml" is working as intended and then we are changing the behavior.

Reply:
Sounds good. Done.